### ISSUE TITLE
Looking for backscattering of nonfiducial events

### ISSUE THREAD

**Issue by DuncanWilmot** [IncandelaLab/LDMX-scripts#27]
## Tasks

- [ ] Understand how to define fiducial and nonfiducial events for v14 8 GeV samples and moving forward
- [ ] Follow nonfiducial events (skim all fiducial events out of the root data) and determine if showers are scattering back into the ECal and depositing energy 

## Background Information
* Speaking generally, nonfiducial events are those in which the recoil electron misses the ECal, or is beyond the "acceptance" of the ECal
* We may find that a real dark matter signature in nature corresponds partially or entirely to nonfiducial events in LDMX, so understanding this subset is important

## Existing material to start off

 * Old code defining fiducial events (make sure to understand how recoil variables are computed kinematically): https://github.com/IncandelaLab/LDMX-scripts/blob/master/EcalVeto/bdtTreeMaker.py#L352-L365
 * Example script with event skimming using uproot (this also writes skimmed data to new root files, which is useful but not needed for this issue) https://github.com/p-masterson/LDMX-scripts/blob/lazy-loading/GraphNet/file_processor.py#L147-L172

**Comment by tvami** [comment]
@DuncanWilmot I found this branch in ldmx-sw, it seems David tried to have it all in ldmx-sw, but maybe it's all outdated at this point:
https://github.com/LDMX-Software/ldmx-sw/compare/nonfiducial-filter?expand=1
We should think about what to do with this branch

(also the sub-branch: https://github.com/LDMX-Software/SimCore/compare/nonfiducial-filter?expand=1

**Comment by DuncanWilmot** [comment]
@tvami Right, I recall David generating nonfiducial samples with this. I will try to find some old overleaf documentation or slides to see if this ended up producing good results (at least for older versions)

**Comment by tvami** [comment]
Sample I mentioned today
```
/home/vamitamas/NonFiducialSimu/events_nonfiducial_test_production.root
```

**Comment by tvami** [comment]
I'll have to redo the sample, but the prod efficiency is veeeery low (0.02%):
```
[ Simulator ] : Started 487736 events to produce 86 events.
```
I'll let you know when I have about 2500 events to look at

**Comment by tvami** [comment]
Here is the final sample stat
```
Started 13366829 events to produce 1897 events.
```
Located at
```
/home/vamitamas/NonFiducialSimu/events_nonfiducial_fullEcal_production.root
```
@51M0nS55 @DuncanWilmot please have a look at it! Thanks

**Comment by tvami** [comment]
The PR about the filter is merged:

https://github.com/LDMX-Software/ldmx-sw/pull/1285

It is named NonFiducialFilter and one example on how to run it can be found at 

https://github.com/LDMX-Software/ldmx-sw/blob/d564390aee82b6f5666574d3f7a8ccf6c2f11d1e/Biasing/test/ecal_pn_nonfiducial.py
